code/parse/sexp.cpp

```diff
@@ -1762,6 +1762,16 @@ int check_operator_argument_count(int count, int op)
 	return 1;
 }
 
+template <typename T>
+int count_named_items(const char *name, int num, const T* item)
+{
```

Comment:
Optional: Validate inputs via assertions, i.e., that the size > 0 and that both pointer arguments are non-null.